data-raw/process_ecmwf_mars_gribs_to_cogs.py

```diff
@@ -0,0 +1,131 @@
+import os
+import tempfile
+from pathlib import Path
+
+import numpy as np
+import pandas as pd
+import rasterio
+import xarray as xr
+from affine import Affine
+from rasterio.crs import CRS
+from scipy.interpolate import griddata
+
+from src.blob_utils import (
+    download_file,
+    list_blobs,
+    load_env_vars,
+    upload_file,
+)
+
+sas_token, container_name, storage_account = load_env_vars()
+
+
+grib_files = list_blobs(
+    sas_token=sas_token,
+    container_name=container_name,
+    storage_account=storage_account,
+    name_starts_with="ds-aa-afg-drought/raw",
+)
+
+grib_files = [file for file in grib_files if file.endswith(".grib")]
+grib_file_samp = grib_files[:1]
+# Loop through each blob path
+for blob_name in grib_file_samp:
+    print(f"name: {os.path.basename(blob_name)}")
+
+    # Create a temporary file
+    with tempfile.TemporaryDirectory() as td:
+        temp_base_path = os.path.basename(blob_name)
+        td = Path(td)
+        tf = td / temp_base_path
+        # Download the file
+        download_file(
+            sas_token=sas_token,
+            container_name=container_name,
+            storage_account=storage_account,
+            blob_path=blob_name,
+            local_file_path=tf,
+        )
+
+        # Open the file with xarray
+        ds = xr.open_dataset(
```

Comment:
In case you want to switch to using lat and lon as dimensions in the dataset, you can do it by converting it into a dataframe then back into a dataset (xarray can't do it for two dimensions at once):
```
df = ds.to_dataframe().reset_index()
df_grouped = df.groupby(
    ["number", "time", "forecastMonth", "latitude", "longitude"]
)["tprate"].first()
ds_from_df = df_grouped.to_xarray()
```
Tbh, this isn't as useful for this case, cause we have to mess around afterwards doing the interpolation anyways. It does however allow for easy plotting (see comment below)

Reply:
oh  thats cool - so any variable that you  `.groupby()`  will become a dimension when cast back `.to_xarray()`? that does seem very useful.

Is the `.first()` call just returning the the first value for use in plotting?